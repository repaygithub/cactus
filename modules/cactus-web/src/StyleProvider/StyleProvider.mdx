---
name: StyleProvider
menu: Components
---

import PropsTable from 'website-src/components/PropsTable'
import { livePreviewStyle } from '../helpers/constants'
import StyleProvider from './StyleProvider'
import cactusTheme from '@repay/cactus-theme'

# StyleProvider

The `StyleProvider` is a wrapper for `styled-components`' `ThemeProvider` component which gives you the option to include some global styles.

# Best practices

- Make sure that any components from `@repay/cactus-web` are children of `StyleProvider`. This will ensure that each component has the correct styles.

## Basic usage

```jsx
import React from 'react'
import { StyleProvider } from '@repay/cactus-web'
import cactusTheme from '@repay/cactus-theme'

const App = () => {
  return (
    <StyleProvider theme={cactusTheme} global={true}>
      // Include routes/rest of application here
    </StyleProvider>
  )
}
```

# Breakpoints

The `StyleProvider` has a media queries scale which is based on the provided breaking points and is added to the given theme. The media queries have screen sizes small, medium, large, and extraLarge. The set media queries will be larger than the default, for a mobile first approach.

Props can also vary depending on the current media query. This is done by setting a prop to an array, where the array holds different values for each media query size. The first element of the array maps to the default screen size, and each element after maps to the set media queries from smallest to largest. This is used below in the example code to set the background color.

For more information on Breakpoints and Media Queries visit: https://styled-system.com/theme-specification#breakpoints

## Media Queries Usage

```jsx
import React from 'react'
import { StyleProvider, Box } from '.@repay/cactus-web'
import styled from 'styled-components'

const BreakpointBox = styled(Box)`
  border-radius: 15%;

  ${p => p.theme.mediaQueries.small} {
    border-radius: 25%;
  }
  ${p => p.theme.mediaQueries.medium} {
    border-radius: 35%;
  }
  ${p => p.theme.mediaQueries.large} {
    border-radius: 45%;
  }
  ${p => p.theme.mediaQueries.extraLarge} {
    border-radius: 50%;
  }
`
const backgroundColors = ['springgreen', 'mediumturquoise', 'cornflowerblue', 'thistle', 'pink']

const App = () => {
  return (
    <StyleProvider theme={cactusTheme} global={true}>
      <BreakpointBox backgroundColor={backgroundColors} width="100px" height="100px" />
    </StyleProvider>
  )
}
```

# Properties

<PropsTable of={StyleProvider} staticProp="StyleProvider" />
