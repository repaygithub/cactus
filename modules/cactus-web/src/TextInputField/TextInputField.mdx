---
name: TextInputField
menu: Components
---

import PropsTable from 'website-src/components/PropsTable'
import TextInputField from './TextInputField'
import cactusTheme from '@repay/cactus-theme'

# TextInputField

The `TextInputField` is a combination of `TextInput`, `Label`, and some handy tools to make integrating this filed into a form easier.

## Best practices

- Use descriptive labels for the input field.
- Be descriptive with the tooltip, if necessary.
- Make use of the `success`, `warning`, and `error` message props when appropriate to make the validity of inputs obvious to the user.

## Basic usage


```jsx
import React, { useState } from 'react'
import { Button, TextInputField } from '@repay/cactus-web'

export default function Form() {
  const [values, setValues] = useState({ input: '' })
  const handleChange = useCallback(
    (name, value) => {
      setValues(state => setValues({ ...state, name: value }))
    },
    [setValues]
  )
  const onSubmit = useCallback(
    event => {
      event.preventDefault()
      const data = { ...values }
      // send data to api
    },
    [values]
  )

  return (
    <form onSubmit={onSubmit}>
      <TextInputField
        name="input"
        label="Input"
        onChange={handleChange}
        value={values.input}
      />
      <Button type="submit" variant="action">
        Submit
      </Button>
    </form>
  )
}

```

## Properties

<PropsTable of={TextInputField} />
