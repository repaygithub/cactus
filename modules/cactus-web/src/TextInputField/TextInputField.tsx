import React, { useRef } from 'react'

import { FieldOnChangeHandler, Omit } from '../types'
import { Label, LabelProps } from '../Label/Label'
import { MarginProps, margins } from '../helpers/margins'
import { TextInput, TextInputProps } from '../TextInput/TextInput'
import StatusMessage, { Status } from '../StatusMessage/StatusMessage'
import styled from 'styled-components'
import Tooltip from '../Tooltip/Tooltip'
import useId from '../helpers/useId'

interface TextInputFieldProps extends MarginProps, Omit<TextInputProps, 'status' | 'onChange'> {
  label: string
  name: string
  labelProps?: LabelProps
  success?: string
  warning?: string
  error?: string
  tooltip?: string
  onChange?: FieldOnChangeHandler<string>
}

const TextInputFieldBase = (props: TextInputFieldProps) => {
  const {
    label,
    labelProps,
    className,
    success,
    warning,
    error,
    tooltip,
    onChange,
    name,
    id,
    ...inputProps
  } = props

  const ref = useRef<HTMLDivElement | null>(null)

  let status: Status | null = null
  if (success && !warning && !error) {
    status = 'success'
  } else if (warning && !success && !error) {
    status = 'warning'
  } else if (error && !success && !warning) {
    status = 'error'
  }

  const inputId = useId(id)
  const statusId = status ? `${inputId}-status` : ''
  const tipId = tooltip ? `${inputId}-tip` : ''

  const handleChange = React.useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      if (typeof onChange === 'function') {
        const currentTarget = (event.currentTarget as unknown) as HTMLInputElement
        onChange(name, currentTarget.value)
      }
    },
    [onChange, name]
  )

  let containerWidth = undefined
  if (ref.current) {
    containerWidth = `${ref.current.getBoundingClientRect().width - 32}px`
  }

  return (
    <div className={className} ref={ref}>
      <Label htmlFor={inputId} {...labelProps}>
        {label}
      </Label>
      {tooltip && <Tooltip id={tipId} label={tooltip} maxWidth={containerWidth} />}
      <TextInput
        {...inputProps}
        id={inputId}
        width="100%"
        status={status}
        onChange={handleChange}
        aria-describedby={`${tipId} ${statusId}`}
      />
      {status === 'success' && (
        <StatusMessage status="success" id={statusId}>
          <span>{success}</span>
        </StatusMessage>
      )}
      {status === 'warning' && (
        <StatusMessage status="warning" id={statusId}>
          <span>{warning}</span>
        </StatusMessage>
      )}
      {status === 'error' && (
        <StatusMessage status="error" id={statusId}>
          <span>{error}</span>
        </StatusMessage>
      )}
    </div>
  )
}

export const TextInputField = styled(TextInputFieldBase)`
  position: relative;
  width: ${p => p.width || 'auto'};

  ${Label} {
    position: relative;
    bottom: 4px;
    padding-left: 15px;
  }

  ${Tooltip} {
    position: absolute;
    right: 8px
    font-size: 16px;
  }

  ${StatusMessage} {
    margin-top: 4px;
  }

  ${margins}
`

TextInputField.defaultProps = {
  error: undefined,
  labelProps: {},
  success: undefined,
  tooltip: undefined,
  warning: undefined,
}

export default TextInputField
