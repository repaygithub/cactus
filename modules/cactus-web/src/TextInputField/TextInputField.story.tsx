import { actions } from '@storybook/addon-actions'
import { boolean, text } from '@storybook/addon-knobs'
import { Meta } from '@storybook/react/types-6-0'
import React, { useState } from 'react'

import TextInputField from './TextInputField'

export default {
  title: 'TextInputField',
  component: TextInputField,
} as Meta

const eventLoggers = actions('onChange', 'onFocus', 'onBlur')

const InputValidator = (): React.ReactElement => {
  const [input, setInput] = useState('')

  const success = input.length > 5 ? 'Your input was successful' : undefined

  return (
    <TextInputField
      name="input"
      label="Type Something"
      tooltip="You must type more than 5 characters"
      onChange={(_, value): void => setInput(value)}
      success={success}
    />
  )
}

export const BasicUsage = (): React.ReactElement => (
  <TextInputField
    label={text('label', 'Input Label')}
    placeholder={text('placeholder', 'Placeholder')}
    disabled={boolean('disabled', false)}
    success={text('success', '')}
    warning={text('warning', '')}
    error={text('error', '')}
    tooltip={text('tooltip', 'Enter some text')}
    name="input-1"
    autoTooltip={boolean('autoTooltip', true)}
    disableTooltip={boolean('disableTooltip', false)}
    {...eventLoggers}
  />
)

BasicUsage.parameters = { cactus: { overrides: { height: '110vh', width: '110vw' } } }

export const FixedWidthContainer = (): React.ReactElement => (
  <div style={{ width: '235px' }}>
    <TextInputField
      label={text('label', 'Input Label')}
      placeholder={text('placeholder', 'Placeholder')}
      disabled={boolean('disabled', false)}
      error={text(
        'error',
        'The input you have entered is unequivocally invalid because we absolutely do not support the information you have provided.'
      )}
      tooltip={text('tooltip', 'Enter some text')}
      name="input-2"
      {...eventLoggers}
    />
  </div>
)

export const Accessibility = (): React.ReactElement => <InputValidator />
