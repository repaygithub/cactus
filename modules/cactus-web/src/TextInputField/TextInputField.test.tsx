import * as React from 'react'

import { cleanup, render } from 'react-testing-library'
import { StyleProvider } from '../StyleProvider/StyleProvider'
import cactusTheme from '@repay/cactus-theme'
import TextInputField from './TextInputField'
import userEvent from 'user-event'

afterEach(cleanup)

describe('component: TextInputField', () => {
  test('should render a TextInputField', () => {
    const { container } = render(
      <StyleProvider>
        <TextInputField name="instigate" label="Enter some text" tooltip="Go on, do it" />
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render a disabled TextInputField', () => {
    const { container } = render(
      <StyleProvider>
        <TextInputField name="trick" label="Come on, type something" tooltip="Sike!" disabled />
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render a TextInputField with a placeholder', () => {
    const { container } = render(
      <StyleProvider>
        <TextInputField
          name="promise"
          label="Ok ok, now do it"
          placeholder="I won't disable it again, promise"
        />
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render a success TextInputField', () => {
    const { container } = render(
      <StyleProvider>
        <TextInputField
          name="success"
          label="No seriously, type something"
          success="Great! you typed something!"
        />
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render a warning TextInputField', () => {
    const { container } = render(
      <StyleProvider>
        <TextInputField name="warn" label="Do it again" warning="Really? That's all you got?" />
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render an error TextInputField', () => {
    const { container } = render(
      <StyleProvider>
        <TextInputField name="error" label="Try again" error="That's it, we're done here" />
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should support margin space props', () => {
    const { container } = render(
      <StyleProvider>
        <TextInputField name="margins" label="Check out all these sick margins" m={3} />
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should trigger onChange handler', () => {
    const onChange = jest.fn()
    const { getByPlaceholderText } = render(
      <StyleProvider>
        <TextInputField
          name="change"
          label="Better type something this time"
          placeholder="Type here!"
          onChange={onChange}
        />
      </StyleProvider>
    )

    userEvent.type(getByPlaceholderText('Type here!'), "Alright fine I'm typing, jeez")
    expect(onChange).toHaveBeenCalled()
  })
})
