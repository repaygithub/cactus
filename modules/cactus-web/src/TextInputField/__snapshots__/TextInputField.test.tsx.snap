// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: TextInputField should render a TextInputField 1`] = `
.c4 {
  font-size: 18px;
  line-height: 28px;
  font-weight: 600;
  color: hsl(200,10%,20%);
}

.c8 {
  border: 2px solid hsl(200,10%,20%);
  border-radius: 20px;
  height: 32px;
  outline: none;
  box-sizing: border-box;
  padding: 7px 28px 7px 15px;
  font-size: 18px;
  line-height: 28px;
  width: 100%;
  background-color: hsl(0,0%,100%);
}

.c8:focus {
  border-color: hsl(200,96%,35%);
}

.c8::-webkit-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c8::-moz-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c8:-ms-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c8::placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c7 {
  box-sizing: border-box;
}

.c2 + .c1 {
  margin-top: 16px;
}

.c6 {
  vertical-align: middle;
  color: hsl(200,96%,35%);
}

.c0 {
  position: relative;
  width: auto;
}

.c0 .c3 {
  position: relative;
  margin-bottom: 4px;
  padding-left: 16px;
}

.c0 .c5 {
  position: absolute;
  top: -2px;
  right: 8px;
  font-size: 16px;
}

.c0 .c9 {
  margin-left: 16px;
  margin-top: 4px;
}

<div>
  <div
    class="c0 c1 c2"
  >
    <label
      class="c3 c4"
      for="instigate"
    >
      Enter some text
    </label>
    <span
      aria-describedby="tooltip:1"
      class="c5 "
      data-reach-tooltip-trigger=""
    >
      <svg
        class="c6"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M22,12 C22,6.47714622 17.5223159,2 12,2 C6.47523326,2 2,6.47577095 2,12 C2,17.524229 6.47523326,22 12,22 C17.5223159,22 22,17.5228538 22,12 Z M24,12 C24,18.6274771 18.6268316,24 12,24 C5.37060996,24 0,18.6287448 0,12 C0,5.37125524 5.37060996,0 12,0 C18.6268316,0 24,5.37252293 24,12 Z M11.057,15.6442233 L11.057,12.5 L9.27493729,12.5 C8.72265254,12.5 8.27493729,12.0522847 8.27493729,11.5 C8.27493729,10.9477153 8.72265254,10.5 9.27493729,10.5 L12.057,10.5 C12.6092847,10.5 13.057,10.9477153 13.057,11.5 L13.057,15.6442233 L15.0505322,15.6442233 C15.602817,15.6442233 16.0505322,16.0919386 16.0505322,16.6442233 C16.0505322,17.1965081 15.602817,17.6442233 15.0505322,17.6442233 L9,17.6442233 C8.44771525,17.6442233 8,17.1965081 8,16.6442233 C8,16.0919386 8.44771525,15.6442233 9,15.6442233 L11.057,15.6442233 Z M11.5,9 C10.6715729,9 10,8.32842712 10,7.5 C10,6.67157288 10.6715729,6 11.5,6 C12.3284271,6 13,6.67157288 13,7.5 C13,8.32842712 12.3284271,9 11.5,9 Z"
        />
      </svg>
    </span>
    <div
      id="instigate-tip"
      role="tooltip"
      style="border: 0px; height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; position: absolute;"
    >
      Go on, do it
    </div>
    <div
      class="c7"
    >
      <input
        aria-describedby="instigate-tip "
        class="c8"
        id="instigate"
        name="instigate"
        width="100%"
      />
    </div>
  </div>
</div>
`;

exports[`component: TextInputField should render a TextInputField with a placeholder 1`] = `
.c4 {
  font-size: 18px;
  line-height: 28px;
  font-weight: 600;
  color: hsl(200,10%,20%);
}

.c6 {
  border: 2px solid hsl(200,10%,20%);
  border-radius: 20px;
  height: 32px;
  outline: none;
  box-sizing: border-box;
  padding: 7px 28px 7px 15px;
  font-size: 18px;
  line-height: 28px;
  width: 100%;
  background-color: hsl(0,0%,100%);
}

.c6:focus {
  border-color: hsl(200,96%,35%);
}

.c6::-webkit-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::-moz-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6:-ms-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c5 {
  box-sizing: border-box;
}

.c2 + .c1 {
  margin-top: 16px;
}

.c0 {
  position: relative;
  width: auto;
}

.c0 .c3 {
  position: relative;
  margin-bottom: 4px;
  padding-left: 16px;
}

.c0 .c8 {
  position: absolute;
  top: -2px;
  right: 8px;
  font-size: 16px;
}

.c0 .c7 {
  margin-left: 16px;
  margin-top: 4px;
}

<div>
  <div
    class="c0 c1 c2"
  >
    <label
      class="c3 c4"
      for="promise"
    >
      Ok ok, now do it
    </label>
    <div
      class="c5"
    >
      <input
        aria-describedby=" "
        class="c6"
        id="promise"
        name="promise"
        placeholder="I won't disable it again, promise"
        width="100%"
      />
    </div>
  </div>
</div>
`;

exports[`component: TextInputField should render a disabled TextInputField 1`] = `
.c4 {
  font-size: 18px;
  line-height: 28px;
  font-weight: 600;
  color: hsl(200,10%,20%);
}

.c8 {
  border: 2px solid hsl(0,0%,90%);
  border-radius: 20px;
  height: 32px;
  outline: none;
  box-sizing: border-box;
  padding: 7px 28px 7px 15px;
  font-size: 18px;
  line-height: 28px;
  width: 100%;
  background-color: hsl(0,0%,90%);
}

.c8:focus {
  border-color: hsl(200,96%,35%);
}

.c8::-webkit-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c8::-moz-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c8:-ms-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c8::placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c7 {
  box-sizing: border-box;
}

.c2 + .c1 {
  margin-top: 16px;
}

.c6 {
  vertical-align: middle;
  color: hsl(200,96%,35%);
}

.c0 {
  position: relative;
  width: auto;
}

.c0 .c3 {
  position: relative;
  margin-bottom: 4px;
  padding-left: 16px;
}

.c0 .c5 {
  position: absolute;
  top: -2px;
  right: 8px;
  font-size: 16px;
}

.c0 .c9 {
  margin-left: 16px;
  margin-top: 4px;
}

<div>
  <div
    class="c0 c1 c2"
  >
    <label
      class="c3 c4"
      for="trick"
    >
      Come on, type something
    </label>
    <span
      aria-describedby="tooltip:2"
      class="c5 "
      data-reach-tooltip-trigger=""
    >
      <svg
        class="c6"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M22,12 C22,6.47714622 17.5223159,2 12,2 C6.47523326,2 2,6.47577095 2,12 C2,17.524229 6.47523326,22 12,22 C17.5223159,22 22,17.5228538 22,12 Z M24,12 C24,18.6274771 18.6268316,24 12,24 C5.37060996,24 0,18.6287448 0,12 C0,5.37125524 5.37060996,0 12,0 C18.6268316,0 24,5.37252293 24,12 Z M11.057,15.6442233 L11.057,12.5 L9.27493729,12.5 C8.72265254,12.5 8.27493729,12.0522847 8.27493729,11.5 C8.27493729,10.9477153 8.72265254,10.5 9.27493729,10.5 L12.057,10.5 C12.6092847,10.5 13.057,10.9477153 13.057,11.5 L13.057,15.6442233 L15.0505322,15.6442233 C15.602817,15.6442233 16.0505322,16.0919386 16.0505322,16.6442233 C16.0505322,17.1965081 15.602817,17.6442233 15.0505322,17.6442233 L9,17.6442233 C8.44771525,17.6442233 8,17.1965081 8,16.6442233 C8,16.0919386 8.44771525,15.6442233 9,15.6442233 L11.057,15.6442233 Z M11.5,9 C10.6715729,9 10,8.32842712 10,7.5 C10,6.67157288 10.6715729,6 11.5,6 C12.3284271,6 13,6.67157288 13,7.5 C13,8.32842712 12.3284271,9 11.5,9 Z"
        />
      </svg>
    </span>
    <div
      id="trick-tip"
      role="tooltip"
      style="border: 0px; height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; position: absolute;"
    >
      Sike!
    </div>
    <div
      class="c7"
    >
      <input
        aria-describedby="trick-tip "
        class="c8"
        disabled=""
        id="trick"
        name="trick"
        width="100%"
      />
    </div>
  </div>
</div>
`;

exports[`component: TextInputField should render a success TextInputField 1`] = `
.c4 {
  font-size: 18px;
  line-height: 28px;
  font-weight: 600;
  color: hsl(200,10%,20%);
}

.c10 {
  vertical-align: middle;
}

.c8 {
  border-radius: 0 8px 8px 8px;
  padding: 8px 16px 8px 16px;
  position: relative;
  box-sizing: border-box;
  overflow-wrap: break-word;
  display: inline-block;
  border: 2px solid;
  font-size: 15px;
  line-height: 23px;
  border-color: hsl(145,89%,28%);
}

.c8 .c11,
.c8 .sc-feJyhm,
.c8 .c9 {
  margin-right: 4px;
  vertical-align: -2px;
}

.c6 {
  border: 2px solid hsl(200,10%,20%);
  border-radius: 20px;
  height: 32px;
  outline: none;
  box-sizing: border-box;
  padding: 7px 28px 7px 15px;
  font-size: 18px;
  line-height: 28px;
  width: 100%;
  background-color: hsl(0,0%,100%);
  border-color: hsl(145,89%,28%);
  background: hsla(145,89%,28%,0.2);
}

.c6:focus {
  border-color: hsl(200,96%,35%);
}

.c6::-webkit-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::-moz-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6:-ms-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c5 {
  box-sizing: border-box;
}

.c2 + .c1 {
  margin-top: 16px;
}

.c0 {
  position: relative;
  width: auto;
}

.c0 .c3 {
  position: relative;
  margin-bottom: 4px;
  padding-left: 16px;
}

.c0 .c12 {
  position: absolute;
  top: -2px;
  right: 8px;
  font-size: 16px;
}

.c0 .c7 {
  margin-left: 16px;
  margin-top: 4px;
}

<div>
  <div
    class="c0 c1 c2"
  >
    <label
      class="c3 c4"
      for="success"
    >
      No seriously, type something
    </label>
    <div
      class="c5"
    >
      <input
        aria-describedby=" success-status"
        class="c6"
        id="success"
        name="success"
        width="100%"
      />
    </div>
    <div
      class="c7 c8"
      id="success-status"
    >
      <svg
        class="c9 c10"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
        />
      </svg>
      <span>
        <span>
          Great! you typed something!
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`component: TextInputField should render a warning TextInputField 1`] = `
.c4 {
  font-size: 18px;
  line-height: 28px;
  font-weight: 600;
  color: hsl(200,10%,20%);
}

.c10 {
  vertical-align: middle;
}

.c11 .c12,
.c11 .c9,
.c11 .sc-hrWEMg {
  margin-right: 4px;
  vertical-align: -2px;
}

.c8 {
  border-radius: 0 8px 8px 8px;
  padding: 8px 16px 8px 16px;
  position: relative;
  box-sizing: border-box;
  overflow-wrap: break-word;
  display: inline-block;
  border: 2px solid;
  font-size: 15px;
  line-height: 23px;
  border-color: hsl(47,82%,47%);
}

.c8 .c12,
.c8 .c9,
.c8 .sc-hrWEMg {
  margin-right: 4px;
  vertical-align: -2px;
}

.c6 {
  border: 2px solid hsl(200,10%,20%);
  border-radius: 20px;
  height: 32px;
  outline: none;
  box-sizing: border-box;
  padding: 7px 28px 7px 15px;
  font-size: 18px;
  line-height: 28px;
  width: 100%;
  background-color: hsl(0,0%,100%);
  border-color: hsl(47,82%,47%);
  background: hsla(47,82%,47%,0.2);
}

.c6:focus {
  border-color: hsl(200,96%,35%);
}

.c6::-webkit-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::-moz-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6:-ms-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c5 {
  box-sizing: border-box;
}

.c2 + .c1 {
  margin-top: 16px;
}

.c0 {
  position: relative;
  width: auto;
}

.c0 .c3 {
  position: relative;
  margin-bottom: 4px;
  padding-left: 16px;
}

.c0 .c13 {
  position: absolute;
  top: -2px;
  right: 8px;
  font-size: 16px;
}

.c0 .c7 {
  margin-left: 16px;
  margin-top: 4px;
}

<div>
  <div
    class="c0 c1 c2"
  >
    <label
      class="c3 c4"
      for="warn"
    >
      Do it again
    </label>
    <div
      class="c5"
    >
      <input
        aria-describedby=" warn-status"
        class="c6"
        id="warn"
        name="warn"
        width="100%"
      />
    </div>
    <div
      class="c7 c8"
      id="warn-status"
    >
      <svg
        class="c9 c10"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M1.39339828,12.7071068 C1.00287399,12.3165825 1.00287399,11.6834175 1.39339828,11.2928932 L11.2928932,1.39339828 C11.6834175,1.00287399 12.3165825,1.00287399 12.7071068,1.39339828 L22.6066017,11.2928932 C22.997126,11.6834175 22.997126,12.3165825 22.6066017,12.7071068 L12.7071068,22.6066017 C12.3165825,22.997126 11.6834175,22.997126 11.2928932,22.6066017 L1.39339828,12.7071068 Z M3.51471863,12 L12,20.4852814 L20.4852814,12 L12,3.51471863 L3.51471863,12 Z M12,16 C11.4477153,16 11,15.5522847 11,15 C11,14.4477153 11.4477153,14 12,14 C12.5522847,14 13,14.4477153 13,15 C13,15.5522847 12.5522847,16 12,16 Z M13,12 C13,12.5522847 12.5522847,13 12,13 C11.4477153,13 11,12.5522847 11,12 L11,9 C11,8.44771525 11.4477153,8 12,8 C12.5522847,8 13,8.44771525 13,9 L13,12 Z"
        />
      </svg>
      <span>
        <span>
          Really? That's all you got?
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`component: TextInputField should render an error TextInputField 1`] = `
.c4 {
  font-size: 18px;
  line-height: 28px;
  font-weight: 600;
  color: hsl(200,10%,20%);
}

.c10 {
  vertical-align: middle;
}

.c11 .c9,
.c11 .sc-feJyhm,
.c11 .sc-hrWEMg {
  margin-right: 4px;
  vertical-align: -2px;
}

.c12 .c9,
.c12 .sc-feJyhm,
.c12 .sc-hrWEMg {
  margin-right: 4px;
  vertical-align: -2px;
}

.c8 {
  border-radius: 0 8px 8px 8px;
  padding: 8px 16px 8px 16px;
  position: relative;
  box-sizing: border-box;
  overflow-wrap: break-word;
  display: inline-block;
  border: 2px solid;
  font-size: 15px;
  line-height: 23px;
  border-color: hsl(353,84%,44%);
}

.c8 .c9,
.c8 .sc-feJyhm,
.c8 .sc-hrWEMg {
  margin-right: 4px;
  vertical-align: -2px;
}

.c6 {
  border: 2px solid hsl(200,10%,20%);
  border-radius: 20px;
  height: 32px;
  outline: none;
  box-sizing: border-box;
  padding: 7px 28px 7px 15px;
  font-size: 18px;
  line-height: 28px;
  width: 100%;
  background-color: hsl(0,0%,100%);
  border-color: hsl(353,84%,44%);
  background: hsla(353,84%,44%,0.2);
}

.c6:focus {
  border-color: hsl(200,96%,35%);
}

.c6::-webkit-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::-moz-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6:-ms-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c5 {
  box-sizing: border-box;
}

.c2 + .c1 {
  margin-top: 16px;
}

.c0 {
  position: relative;
  width: auto;
}

.c0 .c3 {
  position: relative;
  margin-bottom: 4px;
  padding-left: 16px;
}

.c0 .c13 {
  position: absolute;
  top: -2px;
  right: 8px;
  font-size: 16px;
}

.c0 .c7 {
  margin-left: 16px;
  margin-top: 4px;
}

<div>
  <div
    class="c0 c1 c2"
  >
    <label
      class="c3 c4"
      for="error"
    >
      Try again
    </label>
    <div
      class="c5"
    >
      <input
        aria-describedby=" error-status"
        class="c6"
        id="error"
        name="error"
        width="100%"
      />
    </div>
    <div
      class="c7 c8"
      id="error-status"
    >
      <svg
        class="c9 c10"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M11.1055728,1.5527864 C11.4740971,0.815737865 12.5259029,0.815737865 12.8944272,1.5527864 L22.8944272,21.5527864 C23.2268777,22.2176875 22.743382,23 22,23 L2,23 C1.25661798,23 0.773122263,22.2176875 1.10557281,21.5527864 L11.1055728,1.5527864 Z M3.61803399,21 L20.381966,21 L12,4.23606798 L3.61803399,21 Z M12,18 C11.4477153,18 11,17.5522847 11,17 C11,16.4477153 11.4477153,16 12,16 C12.5522847,16 13,16.4477153 13,17 C13,17.5522847 12.5522847,18 12,18 Z M13,14 C13,14.5522847 12.5522847,15 12,15 C11.4477153,15 11,14.5522847 11,14 L11,11 C11,10.4477153 11.4477153,10 12,10 C12.5522847,10 13,10.4477153 13,11 L13,14 Z"
        />
      </svg>
      <span>
        <span>
          That's it, we're done here
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`component: TextInputField should support margin space props 1`] = `
.c4 {
  font-size: 18px;
  line-height: 28px;
  font-weight: 600;
  color: hsl(200,10%,20%);
}

.c6 {
  border: 2px solid hsl(200,10%,20%);
  border-radius: 20px;
  height: 32px;
  outline: none;
  box-sizing: border-box;
  padding: 7px 28px 7px 15px;
  font-size: 18px;
  line-height: 28px;
  width: 100%;
  background-color: hsl(0,0%,100%);
}

.c6:focus {
  border-color: hsl(200,96%,35%);
}

.c6::-webkit-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::-moz-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6:-ms-input-placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c6::placeholder {
  color: hsl(200,18%,45%);
  font-style: oblique;
}

.c5 {
  box-sizing: border-box;
}

.c2 + .c1 {
  margin-top: 16px;
}

.c7 .c3 {
  position: relative;
  margin-bottom: 4px;
  padding-left: 16px;
}

.c0 {
  position: relative;
  width: auto;
  margin: 8px;
}

.c0 .c3 {
  position: relative;
  margin-bottom: 4px;
  padding-left: 16px;
}

.c0 .c9 {
  position: absolute;
  top: -2px;
  right: 8px;
  font-size: 16px;
}

.c0 .c8 {
  margin-left: 16px;
  margin-top: 4px;
}

<div>
  <div
    class="c0 c1 c2"
  >
    <label
      class="c3 c4"
      for="margins"
    >
      Check out all these sick margins
    </label>
    <div
      class="c5"
    >
      <input
        aria-describedby=" "
        class="c6"
        id="margins"
        name="margins"
        width="100%"
      />
    </div>
  </div>
</div>
`;
